### ISSUE TITLE
Generating Top-N Recommendations for the Recommendation System

### ISSUE THREAD

**Issue by PurushothamanSrikanth** [Chinmayrane16/DeepRecommender#1]
@Chinmayrane16 Can we have Top-N recommendation as a new feature to the existing one?
We can generate Top-N recommendations from the rating prediction matrix.

This is a feature request 😇

**Comment by Chinmayrane16** [comment]
@PurushothamanSrikanth Yes, it is possible and fairly simple. At inference, we can get the Top-N indexes (movies) from the user's rating vector with the maximum scores.

Do you wish to proceed and make a PR? ;)

**Comment by PurushothamanSrikanth** [comment]
> @PurushothamanSrikanth Yes, it is possible and fairly simple. At inference, we can get the Top-N indexes (movies) from the user's rating vector with the maximum scores.
> 
> 
> 
> Do you wish to proceed and make a PR? ;)

-  That's great. I'm trying that, but I'm not able to do it properly.

**Comment by PurushothamanSrikanth** [comment]
> > @PurushothamanSrikanth Yes, it is possible and fairly simple. At inference, we can get the Top-N indexes (movies) from the user's rating vector with the maximum scores.
> > Do you wish to proceed and make a PR? ;)
> 
> * [ ]  That's great. I'm trying that, but I'm not able to do it properly.

@Chinmayrane16 Are u talking about `out` matrix in [Training_Deep_AE.ipynb] ?(https://github.com/Chinmayrane16/DeepRecommender/blob/master/Training_Deep_AE.ipynb)

![Untitled1](https://user-images.githubusercontent.com/43005013/109119313-8534c980-776a-11eb-8970-0b4aa15cc49b.png)

**Comment by Chinmayrane16** [comment]
@PurushothamanSrikanth Yeah, you could use that. Choose a particular index (user) and take its rating vector and select Top-N maximum indexes.
Map those indexes back to Movie names. :)

**Comment by PurushothamanSrikanth** [comment]
> @PurushothamanSrikanth Yeah, you could use that. Choose a particular index (user) and take its rating vector and select Top-N maximum indexes.
> 
> Map those indexes back to Movie names. :)

Will give that a try
